### PR TITLE
[SYCL] Add sorting APIs for fixed-size private array input

### PR DESCRIPTION
**Pull request by againull**
These PR covers (1), (2), (3) from  https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/proposed/sycl_ext_oneapi_group_sort.asciidoc#functions-with-fixed-size-arrays.

PR doesn't include key/value sorting with fixed-size array input which will be added in a separate PR.

### CODE REVIEW

**Comment by againull** [comment]
tag @andreyfe1

File: sycl/include/sycl/ext/oneapi/properties/property.hpp [hunk 1]
```diff
@@ -207,6 +207,8 @@ enum PropKind : uint32_t {
   SingleTaskKernel = 66,
   IndirectlyCallable = 67,
   CallsIndirectly = 68,
+  InputDataPlacement = 69,
+  OutputDataPlacement = 70,
   // PropKindSize must always be the last value.
   PropKindSize = 69,
```
Comment (dm-vodopyanov) [inline sycl/include/sycl/ext/oneapi/properties/property.hpp:213]:
```suggestion
  PropKindSize = 71,
```

Comment (againull) [reply sycl/include/sycl/ext/oneapi/properties/property.hpp:213]: Oh, how did I miss that, thank you!

**Comment by dm-vodopyanov** [comment]
@andreyfe1 can you please review?

File: sycl/include/sycl/ext/oneapi/experimental/group_helpers_sorters.hpp [hunk 2]
```diff
@@ -36,6 +37,68 @@ namespace sycl {
...
+inline constexpr input_data_placement_key::value_t<
+    group_algorithm_data_placement::striped>
+    input_data_placement_striped;
+
+inline constexpr output_data_placement_key::value_t<
+    group_algorithm_data_placement::blocked>
+    output_data_placement_blocked;
+inline constexpr output_data_placement_key::value_t<
+    group_algorithm_data_placement::striped>
+    output_data_placement_striped;
```
Comment (andreyfe1) [inline sycl/include/sycl/ext/oneapi/experimental/group_helpers_sorters.hpp:78]: Am I right that structures and variables are required for properties and we need to change the spec accordingly? Currently it says only about `input_data_placement` and `output_data_placement`

**Review by andreyfe1 — COMMENTED** [review]
LGTM overall. Thanks!

**Review by dm-vodopyanov — APPROVED** [review]

File: sycl/include/sycl/ext/oneapi/experimental/group_helpers_sorters.hpp (hunk 2 above)
Comment (againull) [reply sycl/include/sycl/ext/oneapi/experimental/group_helpers_sorters.hpp]: Actually, these are not required, I added them just for convenience. Removed them for now, we can add them if needed with doc change.

File: sycl/include/sycl/ext/oneapi/experimental/group_helpers_sorters.hpp [hunk 3]
```diff
@@ -36,6 +37,54 @@ namespace sycl {
...
+                     std::integral_constant<int, static_cast<int>(Placement)>>;
+};
+
+template <group_algorithm_data_placement Placement>
+inline constexpr input_data_placement_key::value_t<Placement>
+    input_data_placement;
+
+template <group_algorithm_data_placement Placement>
+inline constexpr output_data_placement_key::value_t<Placement>
+    output_data_placement;
```
Comment (andreyfe1) [inline sycl/include/sycl/ext/oneapi/experimental/group_helpers_sorters.hpp:64]: We still have `output_data_placement_key` and `input_data_placement_key` in namespace `sycl::ext::oneapi::experimental`.
Is it expected? What to do with that?

Comment (againull) [reply sycl/include/sycl/ext/oneapi/experimental/group_helpers_sorters.hpp:64]: This is expected, those ones are necessary for kernel properties functionality to work.

Comment (andreyfe1) [reply sycl/include/sycl/ext/oneapi/experimental/group_helpers_sorters.hpp:64]: So, we need to change the Sec accordingly, right?

Comment (againull) [reply sycl/include/sycl/ext/oneapi/experimental/group_helpers_sorters.hpp:64]: Yes

**Pull request merged**